src/App.js

```diff
@@ -1,15 +1,18 @@
-import React from "react";
-import { Route, Switch } from "react-router-dom";
+import React, { useContext } from "react";
+import { Route, Switch, Redirect } from "react-router-dom";
 import Login from "./components/Login/index";
 import User from "./components/User/index";
+import { UserContext } from "./UserContext";
 import "./App.css";
 
 const App = () => {
+  const context = useContext(UserContext);
   return (
     <main className="main">
       <Switch>
         <Route path="/" component={User} exact />
```

Comment:
If user is not logged in he should not be able to go to the user page. Add conditional routing.